File: Frontend/real-estate-front/src/types/favorite.ts
import type { Property } from "./property";

/**
 * Favorite entity interface
 */
export interface Favorite {
  id: string;
  userId: string;
  propertyId: string;
  createdAt: string;
  property?: Property; // Associated property data
}

/**
 * Data structure for creating a new favorite
 */
export interface FavoriteCreate {
  userId: string;
  propertyId: string;
}

// FavoriteCountResponse видалено - не підтримується API

File: Backend/RealEstate/RealEstate.DAL/Entities/Favorite.cs
namespace RealEstate.DAL.Entities
{
    public class Favorite
    {
        public Guid Id { get; set; }
        public Guid UserId { get; set; }
        public Guid PropertyId { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public User User { get; set; } = null!;
        public Property Property { get; set; } = null!;
    }
} 